src/control/server/instance_storage_rpc_test.go

```diff
@@ -185,22 +185,7 @@ func TestIOEngineInstance_bdevScanEngine(t *testing.T) {
 			req:    ctlpb.ScanNvmeReq{},
 			rank:   -1,
 			smdRes: defSmdScanRes(),
-			expResp: &ctlpb.ScanNvmeResp{
-				Ctrlrs: proto.NvmeControllers{
-					func() *ctlpb.NvmeController {
-						c := proto.MockNvmeController(2)
-						c.HealthStats = nil
-						c.SmdDevices = []*ctlpb.SmdDevice{
-							{
-								Rank:     uint32(ranklist.NilRank),
-								RoleBits: storage.BdevRoleAll,
-							},
-						}
-						return c
-					}(),
-				},
-				State: new(ctlpb.ResponseState),
-			},
+			expErr: errors.New("nil superblock"),
```

Comment:
I wonder about changing this behavior... Does this have to be fatal? It is just a scan, right? Does the engine have to be joined for that?

Reply:
Discussed in a chat -- the "old" behavior here is a relatively new change, and logically we should not get into a position where the engine has a bad rank but allows a storage scan. This makes sense to me.